src/text_render.rs

```diff
@@ -514,9 +516,9 @@ where
                 left: image.left,
             },
         );
-    }
-
-    let details = atlas.glyph(&cache_key).unwrap();
+        inner.glyphs_in_use.insert(cache_key);
+        inner.glyph_cache.get(&cache_key).unwrap()
```

Comment:
```suggestion
        inner.glyphs_in_use.insert(cache_key);
        inner.glyph_cache.get_or_insert(cache_key, || GlyphDetails {
            width: image.width,
            height: image.height,
            gpu_cache,
            atlas_id,
            top: image.top,
            left: image.left,
        })
```

This avoids the `unwrap` but differs a bit from the original implementation which `.unwraps`. I'll leave it up to you 🙏